prog/postgres/postgres_server_nexus.rb

```diff
@@ -384,6 +384,20 @@ def before_run
       push self.class, frame, "restart"
     end
 
+    if postgres_server.read_replica?
+      already_lagging = postgres_server.timeline_id_is_lagging_set?
+      currently_lagging = postgres_server.timeline_id_is_lagging
+
+      if already_lagging && currently_lagging
+        postgres_server.incr_recycle
+      elsif currently_lagging
+        postgres_server.incr_timeline_id_is_lagging
+        nap 2 * 60
```

Comment:
2 minutes seems too short. Maybe set it to 15 minutes?

Also, @fdr, what do you think this way of detecting broken replication, which can happen if read replica fetches the update before standby and we perform failover? (Not sure if it can happen in any other way). I believe it is OK, but wanted to get your opinion as well.